src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +149,1026 @@ static DamageTable D_80A63CE0 = {
+
+    s32 i;
+    Vec3f pos;
+
+    pos = this->picto.actor.world.pos;
+    pos.y += 30.0f;
+    pos.x += 20.0f * Math_SinS(this->picto.actor.shape.rot.y);
+    pos.z += 20.0f * Math_CosS(this->picto.actor.shape.rot.y);
+
+    sSparkleAcc.y = -0.4f;
+    sSparkleVel.y = 5.5f;
+
+    for (i = 0; i < 20; i++) {
+        sSparkleVel.x = Rand_Centered() * 3.0f;
+        sSparkleVel.z = Rand_Centered() * 3.0f;
+        EffectSsKirakira_SpawnDispersed(play, &pos, &sSparkleVel, &sSparkleAcc, &sSparkleColorPrim, &sSparkleColorEnv,
+                                        2500, 40);
+    }
+}
+
+void EnBal_EmitDustPuff(EnBal* this, PlayState* play) {
+    static Vec3f sDustVel = { 0.0f, 0.0f, 0.0f };
+    static Vec3f sDustAcc = { 0.0f, 0.0f, 0.0f };
+    static Color_RGBA8 sDustColorPrim = { 255, 255, 255, 255 };
+    static Color_RGBA8 sDustColorEnv = { 198, 198, 198, 255 };
+
+    Vec3f pos = this->picto.actor.world.pos;
+
+    pos.y += 40.0f;
+    sDustVel.x = Rand_CenteredFloat(10.0f);
+    sDustVel.z = Rand_CenteredFloat(10.0f);
+    sDustVel.y = Rand_ZeroFloat(5.0f);
+    func_800B0EB0(play, &pos, &sDustVel, &sDustAcc, &sDustColorPrim, &sDustColorEnv, 150, 40, 15);
+}
+
+// Due to the use of static local variables in the two preceding functions, these
+// static data fields need to be declared/defined down here to match.
+static s32 sGetItemPending = 0;
+static Vec3f sFocusPosMultiplier = { 1000.0f, 0.0f, 0.0f };
+static TexturePtr sEyeTextures[] = { gTingleEyeOpenTex, gTingleEyeClosedTex };
+
+s32 EnBal_CheckIfMapUnlocked(EnBal* this, PlayState* play) {
+    this->purchaseMapId = sBuyMapOptions[this->locationMapId][play->msgCtx.choiceIndex];
+    switch (this->purchaseMapId) {
+        case TINGLE_MAP_CLOCK_TOWN:
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_BOUGHT_TINGLE_MAP_CLOCK_TOWN)) {
+                return true;
+            }
+            break;
+
+        case TINGLE_MAP_WOODFALL:
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_BOUGHT_TINGLE_MAP_WOODFALL)) {
+                return true;
+            }
+            break;
+
+        case TINGLE_MAP_SNOWHEAD:
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_BOUGHT_TINGLE_MAP_SNOWHEAD)) {
+                return true;
+            }
+            break;
+
+        case TINGLE_MAP_ROMANI_RANCH:
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_BOUGHT_TINGLE_MAP_ROMANI_RANCH)) {
+                return true;
+            }
+            break;
+
+        case TINGLE_MAP_GREAT_BAY:
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_BOUGHT_TINGLE_MAP_GREAT_BAY)) {
+                return true;
+            }
+            break;
+
+        case TINGLE_MAP_STONE_TOWER:
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_BOUGHT_TINGLE_MAP_STONE_TOWER)) {
+                return true;
+            }
+            break;
+    }
+    return false;
+}
+
+void EnBal_UnlockSelectedAreaMap(EnBal* this) {
+
+    Inventory_SetWorldMapCloudVisibility(this->purchaseMapId);
```

Comment:
can you remove this empty line?
```suggestion
void EnBal_UnlockSelectedAreaMap(EnBal* this) {
    Inventory_SetWorldMapCloudVisibility(this->purchaseMapId);
```